Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H

```diff
@@ -0,0 +1,190 @@
+    // to use N>1 (for now).
+            const Real x = xlo_global + j*dx;
+#if (AMREX_SPACEDIM == 3)
+            const int l = iv[2];
+            const Real y = ylo_global + k*dy;
+            const Real z = zlo_global + l*dz;
+#else
+            const int l = 0;
+            const int y = 0;
+            const Real z = ylo_global + k*dy;
+#endif
+            for (int n = 0; n < N; ++n)
+            {
+                create1(dst1_data, N*p_offsets[i] + dst1_index + n, x, y, z);
+                create2(dst2_data, N*p_offsets[i] + dst2_index + n, x, y, z);
+            }
+            transform(dst1_data, dst2_data, N*p_offsets[i] + dst1_index,
+                    N*p_offsets[i] + dst2_index, N, arrNumPartCreation(j,k,l));
+        }
+    });
+
+    Gpu::streamSynchronize();
+    return num_added;
+}
+
+/**
+ * \brief Apply a filter on a list of FABs, then create and apply a transform
+ * operation to the particles depending on the output of the filter.
+ *
+ * \tparam N number of particles created in the dst(s) in each cell
+ * \tparam DstTile the dst particle tile type
+ * \tparam FABs the src array of Array4 type
+ * \tparam Index the index type, e.g. unsigned int
+ * \tparam FilterFunc the filter function type
+ * \tparam CreateFunc1 the create function type for dst1
+ * \tparam CreateFunc2 the create function type for dst2
+ * \tparam TransFunc the transform function type
+ *
+ * \param[in,out] dst1 the first destination tile
+ * \param[in,out] dst2 the second destination tile
+ * \param[in] box the box where the particles are created
+ * \param[in] src_FABs An Array of Array4 (e.g. EM fields) defined on box on which
+ *            the filter operation is applied/
+ * \param[in] dst1_index the location at which to starting writing the result to dst1
+ * \param[in] dst2_index the location at which to starting writing the result to dst2
+ * \param[in] filter a callable returning a value > 0 if particles are to be created
+ *            in the considered cell.
+ * \param[in] create1 callable that defines what will be done for the create step for dst1.
+ * \param[in] create2 callable that defines what will be done for the create step for dst2.
+ * \param[in] transform callable that defines the transformation to apply on dst1 and dst2.
+ *
+ * \return num_added the number of particles that were written to dst1 and dst2.
+ */
+template <int N, typename DstTile, typename FABs, typename Index,
+          typename FilterFunc, typename CreateFunc1, typename CreateFunc2,
+           typename TransFunc>
+Index filterCreateTransformFromFAB (DstTile& dst1, DstTile& dst2, const amrex::Box box,
+                                const FABs& src_FABs, const Index dst1_index,
+                                const Index dst2_index, FilterFunc&& filter,
+                                CreateFunc1&& create1, CreateFunc2&& create2,
+                                TransFunc && transform) noexcept
+{
```

Comment:
To be honest I'm not completely sure why we have the overload so I wouldn't know what to add as a comment for that (initially I wrote a single function but subsequently decided to keep the overload to remain consistent with `filterCopyTransform`). I guess it might be useful if we want to call the function using an externally obtained mask rather than a filter function.

I'll add a few comments in front of the for loops.

Reply:
@lucafedeli88 @atmyers could you help us understand the purpose of this overload?

Reply:
I am not completely sure either . However, I agree with @NeilZaim : this implementation leaves the freedom to choose between providing a filter function or a pre-computed mask...

Reply:
I've added a few comments regarding the overloads (also in `FilterCopyTransform.H`) and I've also added a few comments regarding the for loops.

Reply:
For `filterCopyTransformParticles`, we made there be two versions of this function, in case someone already had a pre-computed mask and wanted to use that as the filter instead of a functor. The functor version just computes it's own mask and calls the other overload. That's what the ParalleFor is doing. 

I think it could potentially be useful to have both versions of these, but don't feel too strongly about it.